src/town_gui.cpp

```diff
@@ -1724,22 +1733,52 @@ struct BuildHouseWindow : public PickerWindow {
 		}
 	}
 
+	void OnClick([[maybe_unused]] Point pt, WidgetID widget, [[maybe_unused]] int click_count) override
+	{
+		switch (widget) {
+			case WID_BH_PROTECT_OFF:
+			case WID_BH_PROTECT_ON:
+				this->protect = (widget == WID_BH_PROTECT_ON);
+				this->SetWidgetLoweredState(WID_BH_PROTECT_OFF, !this->protect);
+				this->SetWidgetLoweredState(WID_BH_PROTECT_ON, this->protect);
+
+				if (_settings_client.sound.click_beep) SndPlayFx(SND_15_BEEP);
+				this->SetDirty();
+				break;
+
+			default:
+				this->PickerWindow::OnClick(pt, widget, click_count);
+				break;
+		}
+	}
+
 	void OnInvalidateData(int data = 0, bool gui_scope = true) override
 	{
 		this->PickerWindow::OnInvalidateData(data, gui_scope);
 		if (!gui_scope) return;
 
+		const HouseSpec *spec = HouseSpec::Get(HousePickerCallbacks::sel_type);
+
 		if ((data & PickerWindow::PFI_POSITION) != 0) {
-			const HouseSpec *spec = HouseSpec::Get(HousePickerCallbacks::sel_type);
 			UpdateSelectSize(spec);
 			this->house_info = GetHouseInformation(spec);
 		}
+
+		/* If house spec already has the protected flag, handle it automatically and disable the buttons. */
+		bool protect = HasFlag(spec->extra_flags, BUILDING_IS_PROTECTED);
+		if (protect) this->protect = true;
```

Comment:
Having a member variable and local variable with the same name should be avoided.